docs/source/download.rst

```diff
@@ -1,64 +1,103 @@
+===========================
+Download and Install Devito
+===========================
+
+There are three main approaches to installing Devito.
+
+For those looking for a quick “out-of-the-box” way to try Devito, we recommend
+the route of `Docker installation`_.
+For those interested in running on CPUs, please follow `Docker for CPUs`_.
+For those interested in deploying on GPUs, please follow `Docker for GPUs`_.
+For those looking to use Devito alongside other packages as part of a project, we support
+`pip installation`_. If you are looking to develop for Devito, you might benefit from
+following the `conda installation`_  that includes all the bells and whistles we
 recommend when developing for Devito.
 
+
 Docker installation
 -------------------
+
 You can install and run Devito via Docker_:
 
 .. _Docker: https://www.docker.com/  
 
+Docker for CPUs
+```````````````
+
 .. code-block:: shell
 
-   # get the code
-   git clone https://github.com/devitocodes/devito.git
-   cd devito
+   # NOTE (admin-sudo access may be needed) for some steps
+
+   # 1. Pull Devito image
+   docker pull devitocodes/devito:cpu-latest
 
-   # 1. run the tests
-   docker-compose run devito /tests
+   # 2. (Optional but recommended) Run the tests
+   docker run --rm --name testrun 'devitocodes/devito:cpu-latest' pytest tests/test_operator.py
 
-   # 2. start a jupyter notebook server on port 8888
-   docker-compose up devito
+   # 3. Start a Jupyter notebook server on port 8888
+   docker run --rm -it -p 8888:8888 devitocodes/devito:cpu-latest
 
-   # 3. start a bash shell with devito
-   docker-compose run devito /bin/bash
+   # 4. Start a bash shell with Devito
+   docker run --rm -it devitocodes/devito:cpu-latest /bin/bash
 
-The three sample commands above have only been included as illustrations of typical
+These sample commands above have only been included as illustrations of typical
 uses of Devito inside a container. They are not required to be run in that order.
+Please also review these commands within the dockerfile itself:
+`Dockerfile <https://github.com/devitocodes/devito/blob/master/docker/Dockerfile>`_
+
+
+Docker for GPUs
+```````````````
+
+.. code-block:: shell
+
+   # Requirements: Install `Nvidia container toolkit <https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#setting-up-nvidia-container-toolkit>`_
+
+   # NOTE (admin-sudo access may be needed) for some steps
+   # 1. Pull Devito image
+   docker pull devitocodes/devito:gpu-latest
+
+   # 2. (Optional but recommended) Run the tests
+   docker run --gpus all --rm --name testrun 'devitocodes/devito:gpu-latest' pytest tests/test_gpu_openacc.py
+
+   # 3. Start a Jupyter notebook server on port 8888
+   # Not recommended with the GPU docker, please use the `Docker for CPUs` if you aim for the tutorials
+   docker run --gpus all --rm -it -p 8888:8888 devitocodes/devito:gpu-latest
+
+   # 4. Start a bash shell with Devito
```

Comment:
IMHO the "starting a bash" command should be the first of the list and be emphasized more... the other ones are more of shortcuts to execute invidividual commands, but who actually ever use them? yeah one can be useful to quickly run tests, but...

Reply:
sure